### PR TITLE
Updated Applab and Gamelab debugger to work with libraries

### PR DESCRIPTION
**Pull request by jmkulwik**
This includes an update to the version of the JS Interpreter. See: https://github.com/code-dot-org/JS-Interpreter/pull/36

For reference, here are the relevant acorn config options: https://github.com/code-dot-org/JS-Interpreter/blob/dfd1ed36b632952c8a3ff32866dd788c849a7d27/acorn.js#L99

Disabled library step-in on student created libraries.
Added error messaging that uses the library name.

Runtime error message:
image

Parsing error message:
image

## Links

- [jira](https://codedotorg.atlassian.net/browse/STAR-713)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by jmkulwik** [comment]
Oops. Sorry for the force-push. I had accidentally added a commit to the wrong branch.

File: apps/src/lib/tools/jsinterpreter/JSInterpreter.js [hunk 1]
```diff
@@ -40,6 +41,29 @@ function safeStepInterpreter(jsi) {
...
+/**
+ * Checks the code row number and library name to determine whether we are in
+ * the user's code
+ */
+function isInUserCode(userCodeRow, libraryName) {
+  return -1 !== userCodeRow && !libraryName;
+}
+
+/**
+ * Checks if a library name exists in the node
```
Comment (ajpal) [inline apps/src/lib/tools/jsinterpreter/JSInterpreter.js]: Can you change these comments to be a little less what and a little more why/how? ie. it's pretty easy to tell from the name of this function that it checks whether the node is from a library, but a comment about like, what `node.loc` is and how `node.loc.source` relates to libraries would be more useful IMO.

File: apps/src/lib/tools/jsinterpreter/JSInterpreter.js [hunk 2]
```diff
@@ -238,12 +262,29 @@ export default class JSInterpreter {
...
-        this.interpreter.appendCode(options.libraryCode);
+      // Setting libraryAST to null as that is acorn's default value for the
+      // program option
+      let libraryAST = null;
+      if (Array.isArray(options.projectLibraries)) {
+        options.projectLibraries.forEach(library => {
+          try {
+            libraryAST = acorn.parse(library.code, {
+              ecmaVersion: 5,
+              locations: true,
```
Comment (ajpal) [inline apps/src/lib/tools/jsinterpreter/JSInterpreter.js:278]: Can you add a comment about what the `locations` option does?

File: apps/src/lib/tools/jsinterpreter/JSInterpreter.js [hunk 3]
```diff
@@ -790,7 +835,8 @@ export default class JSInterpreter {
         }
       } else {
         if (this.executionError) {
-          this.handleError(inUserCode ? userCodeRow + 1 : undefined);
+          const row = inUserCode || !!libraryName ? userCodeRow + 1 : undefined;
```
Comment (ajpal) [inline apps/src/lib/tools/jsinterpreter/JSInterpreter.js]: Is row a boolean, number, or undefined depending on the values of these variables? This is kind of a complicated expression so a comment may be helpful.

**Review by ajpal — APPROVED** [review]

**Pull request merged**